tests/integration/Subscribers/ImportExport/PersonalDataExporters/NewsletterOpensExporterTest.php

```diff
@@ -0,0 +1,124 @@
+
+    $this->assertIsArray($result);
+    $this->assertArrayHasKey('data', $result);
+    $this->assertEmpty($result['data']);
+    $this->assertArrayHasKey('done', $result);
+    $this->assertTrue($result['done']);
+  }
+
+  public function testExportReturnsData() {
+    $userEmail = 'email@with.clicks';
+    $userAgentName = 'Mozilla/5.0';
+    $this->prepareDataToBeExported($userEmail, $userAgentName);
+    $expectedData = [
+      ['name' => 'Email subject', 'value' => 'Email Subject'],
+      ['name' => 'Timestamp of the open event', 'value' => '2018-01-02 15:16:17'],
+      ['name' => 'User-agent', 'value' => $userAgentName],
+    ];
+
+    $result = $this->exporter->export($userEmail);
+
+    $this->assertIsArray($result);
+    $this->assertArrayHasKey('data', $result);
+    $this->assertCount(1, $result['data']);
+    $this->assertTrue($result['done']);
+    $this->assertArrayHasKey('group_id', $result['data'][0]);
+    $this->assertArrayHasKey('group_label', $result['data'][0]);
+    $this->assertArrayHasKey('item_id', $result['data'][0]);
+    $this->assertArrayHasKey('data', $result['data'][0]);
+    $this->assertSame($expectedData, $result['data'][0]['data']);
+  }
+
+  public function testExportReturnsDataWhenUserAgentIsUnknown() {
+    $userEmail = 'email@with.clicks';
+    $this->prepareDataToBeExported($userEmail);
+
+    $result = $this->exporter->export($userEmail);
+
+    $this->assertIsArray($result['data']);
+    $this->assertCount(1, $result['data']);
+    $this->assertSame($result['data'][0]['data'][2], ['name' => 'User-agent', 'value' => 'Unknown']);
+  }
+
+  protected function prepareDataToBeExported(string $userEmail, string $userAgentName = null) {
+    $subscriber = Subscriber::createOrUpdate([
+      'email' => $userEmail,
+    ]);
+
+    $queue = SendingQueue::createOrUpdate([
+      'newsletter_rendered_subject' => 'Email Subject',
+      'task_id' => 1,
+      'newsletter_id' => 8,
+    ]);
+
+    $newsletter = Newsletter::createOrUpdate([
+      'subject' => 'Email Subject1',
+      'type' => Newsletter::TYPE_STANDARD,
+    ]);
+
+    if ($userAgentName) {
+      $userAgent = new UserAgentEntity($userAgentName);
+      $this->entityManager->persist($userAgent);
+      $this->entityManager->flush();
+      $userAgentId = $userAgent->getId();
+    } else {
+      $userAgentId = null;
+    }
+
+    StatisticsOpens::createOrUpdate([
+      'newsletter_id' => $newsletter->id(),
+      'subscriber_id' => $subscriber->id(),
+      'queue_id' => $queue->id(),
+      'created_at' => '2018-01-02 15:16:17',
+      'user_agent_id' => $userAgentId,
+    ]);
+  }
+
+  public function _after() {
+    $this->truncateEntity(SubscriberEntity::class);
+    $this->truncateEntity(SendingQueueEntity::class);
+    $this->truncateEntity(NewsletterEntity::class);
+    $this->truncateEntity(UserAgentEntity::class);
+    ORM::raw_execute('TRUNCATE ' . StatisticsOpens::$_table);
```

Comment:
I know you assigned this PR to @costasovo, But because he is on vacation, I took a quick look before he can prop

I don't understand why you are using the old models in the tests. Here you could use the same method `truncateEntity` as you did on the previous lines. And also, for creating the data, you used the old models. We shouldn't use the old models in the new code because we are creating new technical debt. Someone in the future would have to come back and do the refactor. I can see your note in the description:

> I meant to refactor the tests to use Doctrine instead of Paris as well but ran out of time.

But I am not sure I understand. What time do you have in mind? Is there a timebox on this feature?

Reply:
> I know you assigned this PR to @costasovo, But because he is on vacation, I took a quick look before he can prop

I opted to assign to Rosta even though he was on vacation as Friday was his last day off, and I'm not working on Fridays. So a review this week was not a problem. I also opted to do this since Jan is on vacation this week, and you are on L2 support. Anyway, thanks for jumping in and reviewing the PR, @pavel-mailpoet.

> I don't understand why you are using the old models in the tests. Here you could use the same method truncateEntity as you did on the previous lines. And also, for creating the data, you used the old models. We shouldn't use the old models in the new code because we are creating new technical debt. Someone in the future would have to come back and do the refactor.

That was a slip on my part. While working on this last week, I was under the false impression that there was not a `StatisticsOpenEntity` class which I can see now is not the case. I believe what happened is that to check for the existence of the class, I used the auto-complete feature of my IDE without realizing that the Paris models use the plural form and the Doctrine models use the singular form. So I was probably looking for `StatisticsOpensEntity`, and that is why I found nothing. I then opted to leave the code as is since I didn't want to change the focus of this task to work on creating a new Doctrine model.

Now that I can see that `StatisticsOpenEntity` exists, I pushed a new commit refactoring NewsletterClicksExporterTest.php and NewsletterOpensExporterTest.php.

> But I am not sure I understand. What time do you have in mind? Is there a timebox on this feature?

There was no timebox for this task. What I meant to say here is that I didn't have enough time to do the refactor (which I was wrongly assuming was a much more involved task than it actually is, as I explained above) before the end of my day. I thought it was not worth delaying this task because of a Doctrine related refactor since the functionality was finished.